docs/theming.md

```diff
@@ -0,0 +1,297 @@
+-->
+
+# Theming Guide
+
+## Overview
+
+Material Web Components (MWC) include a theming system designed to make it easy to customize your application's UI by changing 
+aspects including typography, color, and component shape. MWC supports theming using CSS Custom Properties.
+
+## Theming with CSS custom properties
+
+MWC supports theming using [CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties). 
+Custom properties are set just like standard CSS properties:
+
+```css
+html {
+  --mdc-theme-primary: fuchsia;
+}
+```
+
+In addition to setting custom properties to customize the appearance of MWC components, you can also 
+use them to style your own components (or generic HTML elements):
+
+```css
+div.myCard {
+  background-color: var(--mdc-theme-primary);
+}
+```
+
+### Custom properties and older browsers
+
+CSS custom properties are supported on most modern browsers. However, if you need to support older browsers,
+you'll need to take a few extra steps. (These steps primarily apply to Internet Explorer 11.)
+
+1.  Load the web components polyfills, as described in the [Getting started guide](../README.md#supporting-older-browsers).
+
+2.  Load `CustomStyleInterface` (after loading the web components polyfills):
+
+    ```html
+    <script src="@webcomponents/shadycss/custom-style-interface.min.js"></script>
+    ```
+
+3.  Identify any top-level style elements that define custom properties. Add
+    a class to these style elements so you can easily locate them from script.
+
+    ```html
+    <style class="document-style">
+    html {
+      --mdc-theme-primary: brown;
+    }
+    </style>
+    ```
+
+4.  From script, call `addCustomStyle` for all document level style elements. 
+
+    ```js
+    window.ShadyCSS.CustomStyleInterface.addCustomStyle(document.querySelector('style.document-style'));
+    ```
+
+    This code assumes a single top-level style element. If you have more than one, you'll need to modify the code.
+    See [About CustomStyleInterface](https://github.com/webcomponents/polyfills/tree/master/packages/shadycss#about-customstyleinterface)
+    in the Shady CSS README for more examples of using `CustomStyleInterface`.
+
+With these steps, the polyfills provide limited support for CSS custom properties. 
+See [Limitations](https://github.com/webcomponents/polyfills/tree/master/packages/shadycss#limitations)
+in the ShadyCSS README for details.
+
+## Colors
+
+Material Design theming uses two main colors: **primary** and **secondary**. The primary color is used
+throughout most of the application and components, as the main color for your application. The secondary color is used
+for floating action buttons and other interactive elements, serving as visual contrast to the primary.
+
+In addition to the primary and secondary colors, Material Theming also defines a _background_ color for the app, 
+as well as a _surface_ color, which is used as a background in components.
```

Comment:
What do you think of this addition? I feel like the difference between "background" and "surface" can be a bit confusing. There's only one background, but there are a bunch of surfaces on top of it.
```suggestion
as well as a _surface_ color, which is used as a background in components on top of the app's background.
```